### PR TITLE
Implement Sort feature

### PR DESCRIPTION
**Pull request by theman-oj10**
Sort Command and relevant tests are implemented

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/InternshipComparators.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            StatusEnum.ACCEPTED, 3,
+            StatusEnum.REJECTED, 4
+    );
+
+    /**
+     * Returns a comparator that compares two internships based on the application status.
+     * @param ascending Whether to sort in ascending order.
+     * @return A comparator that compares two internships based on the application status.
+     */
+    public static Comparator<Internship> byApplicationStatus(boolean ascending) {
```
Comment (HM33-Stu) [inline src/main/java/seedu/address/logic/commands/InternshipComparators.java:26]: If Im not mistaken, the convention to follow is: isXXX....
Is it ok to name a boolean as such?

Comment (theman-oj10) [reply src/main/java/seedu/address/logic/commands/InternshipComparators.java:26]: Will fix this in next PR thanks

File: src/main/java/seedu/address/logic/parser/InternshipSortCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, SUPPORTED_PREFIXES);
+        // is there a way to check which prefix was added into argmultimap?
+
+        if (!anyPrefixesPresent(argMultimap, SUPPORTED_PREFIXES)) {
+            logger.warning("Internship sort command has no valid prefixes");
+            throw new ParseException(InternshipSortCommand.MESSAGE_INVALID_FIELD);
+        }
+        OrderEnum parsedOrder = assignOrder(argMultimap);
+        FieldEnum field = assignField(argMultimap);
+        return new InternshipSortCommand(field, parsedOrder);
```
Comment (LWS49) [inline src/main/java/seedu/address/logic/parser/InternshipSortCommandParser.java:136]: if multiple prefixes are present, i assume only the first one is taken while the rest is treated as the string for the first prefix (and therefore invalid_order is thrown)? also based on 145-150 where assignField returns after finding the first prefix

File: src/test/java/seedu/address/logic/commands/InternshipComparatorsTest.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    @Test
+    public void byPhone_descComparator_correctOrder() {
+        Comparator<Internship> comparator = InternshipComparators.byPhone(false);
+        Internship internship1 = internshipBuilder.withContactNumber("12345678").build();
+        Internship internship2 = internshipBuilder.withContactNumber("23456789").build();
+        assertEquals(1, comparator.compare(internship1, internship2));
+    }
+
```
Comment (LWS49) [inline src/test/java/seedu/address/logic/commands/InternshipComparatorsTest.java:83]: byLocation ascOrder?

File: src/test/java/seedu/address/logic/commands/InternshipSortCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        assertTrue(sortCompanyNameCommand.equals(sortNameCommandCopy));
+
+        // different types -> returns false
+        assertFalse(sortCompanyNameCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(sortCompanyNameCommand.equals(null));
+
+        // different field -> returns false
+        assertFalse(sortCompanyNameCommand.equals(sortLocationCommand));
```
Comment (LWS49) [inline src/test/java/seedu/address/logic/commands/InternshipSortCommandTest.java:84]: different order field test

File: src/test/java/seedu/address/logic/commands/InternshipSortCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    @Test
+    public void toString_validFieldAndOrder_success() {
+        InternshipSortCommand sortNameCommand = new InternshipSortCommand(fieldCompany, ascending);
+        String expected = InternshipSortCommand.class.getCanonicalName() + "{field="
+                + fieldCompany.toString() + ", order=" + ascending.toString() + "}";
+        assertEquals(expected, sortNameCommand.toString());
+    }
+}
```
Comment (LWS49) [inline src/test/java/seedu/address/logic/commands/InternshipSortCommandTest.java:128]: add sort by description, role, email, phone, remark
if possible add test to sort such that there's more than one entry, where the sorted list actually has a different order from filtered list (to check that sort actually changes the order properly)

File: src/main/java/seedu/address/logic/parser/InternshipSortCommandParser.java (hunk 2 above)
Comment (theman-oj10) [reply src/main/java/seedu/address/logic/parser/InternshipSortCommandParser.java:136]: Yes this is right

File: src/main/java/seedu/address/model/internship/Internship.java [hunk 6]
```diff
@@ -113,6 +116,31 @@ public boolean isSameInternship(Internship otherInternship) {
...
+            return InternshipComparators.byRole(ascending);
+        case REMARK:
+            return InternshipComparators.byRemark(ascending);
+        case DESCRIPTION:
+            return InternshipComparators.byDescription(ascending);
+        default:
+            throw new IllegalArgumentException("Invalid field for sorting: " + field);
+        }
+    }
+
```
Comment (CaptainKeqing) [inline src/main/java/seedu/address/model/internship/Internship.java:143]: I think this function shouldn't exist under Internship.java

File: src/main/java/seedu/address/model/InternshipModelManager.java [hunk 7]
```diff
@@ -123,9 +127,17 @@ public void setInternship(Internship target, Internship editedPerson) {
      */
     @Override
     public ObservableList<Internship> getFilteredInternshipList() {
-        return filteredInternships;
```
Comment (CaptainKeqing) [inline src/main/java/seedu/address/model/InternshipModelManager.java:126]: If we are now returning sortedInternships then should the name be changed

**Review by CaptainKeqing — COMMENTED** [review]
Just some small nits for you to look into

File: src/test/java/seedu/address/logic/commands/InternshipComparatorsTest.java (hunk 3 above)
Comment (theman-oj10) [reply src/test/java/seedu/address/logic/commands/InternshipComparatorsTest.java:83]: Done

**Comment by jieqiboh** [comment]
Bug: ```sort /poc asc /com asc``` is a valid command which only sorts by ```/com asc``

**Review by jieqiboh — APPROVED** [review]
lgtm!

**Pull request merged**